pkg/storegateway/series_refs_test.go

```diff
@@ -580,6 +587,155 @@ func TestMergedSeriesChunkRefsSet(t *testing.T) {
 	}
 }
 
+func TestMergedSeriesChunkRefsSet_Concurrency(t *testing.T) {
+	const (
+		concurrency        = 10
+		runs               = 100
+		minIterators       = 2
+		maxIterators       = 20
+		minSetsPerIterator = 2
+		maxSetsPerIterator = 10
+		minSeriesPerSet    = 10
+		maxSeriesPerSet    = 50
+	)
+
+	runTest := func() {
+		// Randomize the test setup.
+		var (
+			numIterators       = minIterators + rand.Intn(maxIterators-minIterators)
+			numSetsPerIterator = minSetsPerIterator + rand.Intn(maxSetsPerIterator-minSetsPerIterator)
+			numSeriesPerSet    = minSeriesPerSet + rand.Intn(maxSeriesPerSet-minSeriesPerSet)
+		)
```

Comment:
nit: maybe also add a seed that we print before starting, so we can reproduce a failing test? 

aside: This looks like a good use case for fuzzing, although I think we cannot control the size of the the input for fuzzing and we definitely don't want `numSetsPerIterator=10000000000`

Reply:
> nit: maybe also add a seed that we print before starting, so we can reproduce a failing test?

I thought about it, but I think wouldn't be reproduceable anyway. This test tests concurrency, so you can't reproduce the execution in the same order of a previous run.